### PR TITLE
refactor: Make query() return an Iterable

### LINKED ISSUE #31: refactor: query() should return an Iterable, not List
As discussed [on Discord](https://discord.com/channels/509714518008528896/1255127093310918686/1255170714709921812), returning a `List` from [`QueryableOrderedSet.query()`](https://github.com/bluefireteam/ordered_set/blob/main/lib/queryable_ordered_set.dart#L73C11-L73C16) leads to bugs. The developer assumes they own the returned list, then modify it in place (with `removeWhere()`, for example) to avoid creating yet another iterable. Unbeknownst to the, they're modifying the internal cache of the `QueryableOrderedSet`.

We agreed to make the breaking change of returning `Iterable<C>` on [this line](https://github.com/bluefireteam/ordered_set/blob/main/lib/queryable_ordered_set.dart#L73). Nothing other than that function needs to change, as `List` is already an `Iterable`, so we're only exposing the same data structure as a different static type.

This is breaking because people can have the results of `query()` assigned to typed variables, such as `List<Entity> blah = world.query<Entity>();`. But that's easy to fix.

If anyone is actually _using_ the return of `query()` as a modifiable `List` (as in, they modify it), then that's likely a bug. The change will uncover the bug, so all's good.

The only other valid thing that this could change is code such as `var something = blah[index]`, or `for (int i = 0; i < blah.length; i++) doThing(blah[i]);`. These are probably rare but valid.

cc @spydon

### PR DESCRIPTION
**Pull request by filiph**
Fixes #31

I added a unit test but since it's not really testing anything important, I'm happy to remove it.

I also went ahead and upped semver, though that, too, can be removed from this PR.

cc @spydon

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: ordered_set
 description: A simple implementation of an Ordered Set for Dart that allows multiple items with the same priority.
-version: 5.0.3
-homepage: https://github.com/luanpotter/ordered_set
+version: 6.0.0
+homepage: https://github.com/bluefireteam/ordered_set
```
Comment (filiph) [inline pubspec.yaml:4]: Drive by fix. Unless keeping the homepage as https://github.com/luanpotter/ordered_set was intentional, in which case it's easy to remove this change.

File: test/queryable_ordered_set_test.dart [hunk 2]
```diff
@@ -278,6 +278,26 @@ void main() {
...
+        orderedSet.add(dog);
+        orderedSet.add(bird);
+
+        final dogs = orderedSet.query<Dog>();
+        expect(dogs, isA<Iterable<Dog>>());
+
+        // Note that the following fails, since it _is_ possible to force cast
+        // the returned value as a List.
+        // expect(dogs is List<Dog>, isFalse);
+        // expect(dogs, isNot(isA<List<Dog>>()));
```
Comment (spydon) [inline test/queryable_ordered_set_test.dart]: I think this comment block can be removed

**Review by spydon — APPROVED** [review]
Lgtm! It's fine with the drive-by fix and version bumping in this repo since it's so seldom updated. :)

File: test/queryable_ordered_set_test.dart (hunk 2 above)
Comment (filiph) [reply test/queryable_ordered_set_test.dart]: Done.

**Comment by filiph** [comment]
Thanks for the approval! I think the CI needs an explicit approval before running.

**Pull request merged**